docker/models/containers.py

```diff
@@ -709,7 +716,14 @@ def run(self, image, command=None, stdout=True, stderr=False,
         if exit_status != 0:
             stdout = False
             stderr = True
-        out = container.logs(stdout=stdout, stderr=stderr)
+
+        logging_driver = container.attrs['HostConfig']['LogConfig']['Type']
+
+        if logging_driver == 'json-file' or logging_driver == 'journald':
```

Comment:
I wonder why it's using the logging endpoint, and not attaching to the container (which is what `docker run` does)